src/main/java/com/productdock/library/inventory/application/service/DeleteBookService.java

```diff
@@ -0,0 +1,23 @@
+package com.productdock.library.inventory.application.service;
+
+import com.productdock.library.inventory.application.port.in.DeleteBookUseCase;
+import com.productdock.library.inventory.application.port.out.persistence.InventoryRecordsPersistenceOutPort;
+import com.productdock.library.inventory.domain.exception.InventoryException;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class DeleteBookService implements DeleteBookUseCase {
+
+    private InventoryRecordsPersistenceOutPort inventoryRecordRepository;
+
+    @Override
+    public void deleteBook(String bookId) {
+        var bookInventory = inventoryRecordRepository.findByBookId(bookId).orElseThrow(() -> new InventoryException("Book does not exist in inventory!"));
```

Comment:
What will happen with the Kafka message if we throw an exception here when the book doesn't exist?
Are we going to be stuck in the loop, and if yes, how to solve it?

Reply:
Kafka consumer, by default, will make 10 attempts before breaking the loop.